### PR TITLE
Fix unsound implementation of the RTP monitoring go-routine as well as remove obsolete stuff

### LINKED ISSUE #92: Create the sound implementation of the rtp subscription watchdog
We must monitor the RTP packets on subscriptions and print an error if the subscription does not receive any RTP packets for a long time.

This has 2 purposes:
1. We'll be able to see when the subscription is "stalled" which would explain freezes (in case the freeze occurs, we'll know why).
2. We can use this information to switch simulcast layers on the SFU if we detect that one or another layer is not available, i.e. if a high level is used and becomes unavailable.

### PR DESCRIPTION
**Pull request by daniel-abramov**
This fixes https://github.com/matrix-org/waterfall/issues/92.

### CODE REVIEW

File: pkg/common/watchdog.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		c.channel <- struct{}{}
+		return true
+	}
+
+	return false
+}
+
+// Starts a watchdog that periodically (specified by the configuration) executes a `c.OnTimeout` closure if
+// no messages have been received on a channel for a `c.Timeout`.
+func (c *WatchdogConfig) Start() *WatchdogChannel {
```
Comment (SimonBrandner) [inline pkg/common/watchdog.go:51]: The usage of `WatchdogConfig` here feels odd to me, wouldn't a `NewWatchdogChannel(c WatchdogConfig) *WatchdogChannel {}` be better? Or is this the go way to do things?

File: pkg/common/watchdog.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	// The channel that will be used to inform the watchdog about the reception of a packet.
+	// The watchdog will be stopped once the channel is closed.
+	incoming := make(chan struct{}, UnboundedChannelSize)
+
+	go func() {
+		for {
+			select {
+			case _, ok := <-incoming:
+				if !ok {
+					return
```
Comment (SimonBrandner) [inline pkg/common/watchdog.go:61]: Should we set closed to false, log an error or do something here? Just returning feels dangerous?

**Review by SimonBrandner — APPROVED** [review]
Otherwise LGTM

File: pkg/common/watchdog.go (hunk 1 above)
Comment (daniel-abramov) [reply pkg/common/watchdog.go:51]: We did something similar in `HeartBeat`, so I just followed the same approach, but you're probably right, perhaps it's better to have an explicit `NewWatchdog()` function that accepts the config, let me rework that real quick.

**Review by SimonBrandner — APPROVED** [review]

**Pull request merged**